examples/stm32c0x/adc-simple/main.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2019, Niklas Hauser
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include "../custom-board.hpp"
+
+using namespace Board;
+using namespace std::chrono_literals;
+
+#undef MODM_LOG_LEVEL
+#define MODM_LOG_LEVEL modm::log::DEBUG
+
+// Create an IODeviceWrapper around the Uart Peripheral we want to use
+modm::IODeviceWrapper<proto_uart::Uart, modm::IOBuffer::BlockIfFull> loggerDevice;
+// Set all four logger streams to use the UART
+modm::log::Logger modm::log::debug(loggerDevice);
+modm::log::Logger modm::log::info(loggerDevice);
+modm::log::Logger modm::log::warning(loggerDevice);
+modm::log::Logger modm::log::error(loggerDevice);
+
+// ----------------------------------------------------------------------------
+int
+main()
+{
+	uint16_t vref;
+
+	Board::initialize();
+	Adc1::connect<GpioA1::In1, GpioA2::In2>();
+	Adc1::initialize<Board::SystemClock, Adc1::ClockMode::Asynchronous,
+					 24_MHz>();  // 24MHz/160.5 sample time=6.6us (fulfill Ts_temp of 5us)
+
+	Adc1::setResolution(Adc1::Resolution::Bits12);
+	Adc1::setSampleTime(Adc1::SampleTime::Cycles160_5);
+	Adc1::setRightAdjustResult();
+	Adc1::enableOversampling(Adc1::OversampleRatio::x16, Adc1::OversampleShift::Div16);
+
+	while (true)
+	{
+		LedGn::toggle();
+		modm::delay(500ms);
+
+		vref = Adc1::readInternalVoltageReference();
+		MODM_LOG_INFO << "vref=" << vref << "mV "
+					  << "temperature=" << Adc1::readTemperature(vref) << "deg.C "
+					  << "ch1=" << (vref * Adc1::readChannel(Adc1::Channel::In1) / 4096U) << "mV "
+					  << "ch2=" << (vref * Adc1::readChannel(Adc1::Channel::In2) / 4096U) << "mV "
```

Comment:
Shouldn't the ADC max value rather be 4095?

Reply:
yes, replaced with 0xFFF to make it clearer.